ims/common/config.py

```diff
@@ -32,82 +33,50 @@ class __BMIConfig:
-                                       constants.HAAS_URL_KEY)
+            self.fs = dict(config.items(constants.FS_CONFIG_SECTION_NAME))
+            self.netiso = dict(
+                config.items(constants.NETISO_CONFIG_SECTION_NAME))
+            self.iscsi = dict(config.items(constants.ISCSI_CONFIG_SECTION_NAME))
+            self.tftp = dict(config.items(constants.TFTP_CONFIG_SECTION_NAME))
+            self.bmi = dict(config.items(constants.BMI_CONFIG_SECTION_NAME))
+            self.logs = dict(config.items(constants.LOGS_CONFIG_SECTION_NAME))
+            self.http = dict(config.items(constants.HTTP_CONFIG_SECTION_NAME))
+            self.rpc = dict(config.items(constants.RPC_CONFIG_SECTION_NAME))
+            self.db = dict(config.items(constants.DB_CONFIG_SECTION_NAME))
+
+            if constants.TESTS_CONFIG_SECTION_NAME in config.sections():
+                self.tests = dict(
+                    config.items(constants.TESTS_CONFIG_SECTION_NAME))
 
-            self.nameserver_ip = config.get(constants.RPC_CONFIG_SECTION_NAME,
-                                            constants.RPC_NAME_SERVER_IP_KEY)
-            self.nameserver_port = int(
-                config.get(constants.RPC_CONFIG_SECTION_NAME,
-                           constants.RPC_NAME_SERVER_PORT_KEY))
-            self.rpcserver_ip = config.get(constants.RPC_CONFIG_SECTION_NAME,
-                                           constants.RPC_RPC_SERVER_IP_KEY)
-            self.rpcserver_port = int(
-                config.get(constants.RPC_CONFIG_SECTION_NAME,
-                           constants.RPC_RPC_SERVER_PORT_KEY))
-
-            self.bind_ip = config.get(constants.HTTP_CONFIG_SECTION_NAME,
-                                      constants.BIND_IP_KEY)
-            self.bind_port = config.get(constants.HTTP_CONFIG_SECTION_NAME,
-                                        constants.BIND_PORT_KEY)
-
-            self.logs_url = config.get(constants.LOGS_CONFIG_SECTION_NAME,
-                                       constants.LOGS_URL_KEY)
-            self.logs_debug = config.get(constants.LOGS_CONFIG_SECTION_NAME,
-                                         constants.LOGS_DEBUG_KEY) == 'True'
-            self.logs_verbose = config.get(constants.LOGS_CONFIG_SECTION_NAME,
-                                           constants.LOGS_VERBOSE_KEY) == 'True'
-            self.pxelinux_loc = config.get(constants.TFTP_CONFIG_SECTION_NAME,
-                                           constants.PXELINUX_URL_KEY)
-            self.ipxe_loc = config.get(constants.TFTP_CONFIG_SECTION_NAME,
-                                       constants.IPXE_URL_KEY)
-
-            for k, v in config.items(constants.FILESYSTEM_CONFIG_SECTION_NAME):
-                if v == 'True':
-                    self.fs[k] = {}
-
-                    for key, value in config.items(k):
-                        self.fs[k][key] = value
         # Didn't Test
```

Comment:
If you are concerned about how to test this, the ideal way would be to load a config file that doesn't contain some of mandatory sections.